unlock-app/src/components/interface/locks/CheckoutUrl/elements/DynamicForm.tsx

```diff
@@ -203,13 +203,84 @@ const AddressInputComponent = ({
   )
 }
 
-const ComponentByTypeMap: ComponentByTypeMapProps = {
+const IconInputComponent = ({ name, label, description, onChange }: any) => {
+  const { mutateAsync: uploadImage, isLoading: isUploading } = useImageUpload()
+  const [isOpen, setIsOpen] = useState(false)
+
+  return (
+    <>
+      <div className="flex flex-col gap-1">
+        <Button
+          variant="outlined-primary"
+          size="small"
+          onClick={() => {
+            setIsOpen(true)
+          }}
+        >
+          {label}
+        </Button>
+        {description && (
+          <span className="text-xs text-gray-600">{description}</span>
+        )}
+      </div>
+      <Drawer isOpen={isOpen} setIsOpen={setIsOpen}>
+        <ConnectForm>
+          {({ watch, setValue }: any) => {
+            const image = watch(name) ?? ''
+            const fields = watch()
+
+            return (
+              <div className="grid grid-cols-1">
```

Comment:
Let's make it a form and use `onSubmit` instead of `onClick` on the button. It is better for accessibility!